// @epignosis_llc/gnosis/dist/esm/src/components/Avatar/Avatar.js

  /**
   * @license
   * author: epignosis front-end team
   * @epignosis_llc/gnosis.js v2.2.1
   * Released under the MIT license.
   */

import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import 'react';
import { avatar } from './styles.js';
import { createElement as jsx } from '../../../node_modules/@emotion/react/dist/emotion-react.browser.esm.js';

var _excluded = ["src", "size", "bgColor", "alt", "children"];

var Avatar = function Avatar(_ref) {
  var src = _ref.src,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? "sm" : _ref$size,
      _ref$bgColor = _ref.bgColor,
      bgColor = _ref$bgColor === void 0 ? "#0E59BB" : _ref$bgColor,
      alt = _ref.alt,
      children = _ref.children,
      rest = _objectWithoutProperties(_ref, _excluded);

  return jsx("figure", _extends({
    css: function css(theme) {
      return avatar(theme, {
        size: size,
        bgColor: bgColor
      });
    }
  }, rest), Boolean(src) && jsx("img", {
    src: src,
    alt: alt
  }), children !== undefined && jsx("span", {
    "data-testid": "avatar-children-container"
  }, children));
};

export { Avatar as default };
