// @epignosis_llc/gnosis/dist/esm/src/components/Chip/Chip.js

  /**
   * @license
   * author: epignosis front-end team
   * @epignosis_llc/gnosis.js v2.2.1
   * Released under the MIT license.
   */

import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import 'react';
import SvgClose from '../../icons/close.svg.js';
import { chip } from './styles.js';
import { createElement as jsx } from '../../../node_modules/@emotion/react/dist/emotion-react.browser.esm.js';

var _excluded = ["size", "onClose", "children", "style"];

var Chip = function Chip(_ref) {
  var _ref$size = _ref.size,
      size = _ref$size === void 0 ? "md" : _ref$size,
      onClose = _ref.onClose,
      children = _ref.children,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {
    backgroundColor: "#5c5c5c",
    color: "#fff"
  } : _ref$style,
      rest = _objectWithoutProperties(_ref, _excluded);

  return jsx("div", _extends({
    css: function css(theme) {
      return chip(theme, {
        size: size
      });
    },
    style: style
  }, rest), onClose && jsx("button", {
    onClick: onClose,
    "aria-label": "Remove ".concat(children),
    style: {
      color: style.color
    }
  }, jsx(SvgClose, {
    height: 16
  })), children);
};

export { Chip as default };
