// react-spinners/helpers/proptypes.js

  /**
   * @license
   * author: epignosis front-end team
   * @epignosis_llc/gnosis.js v2.2.1
   * Released under the MIT license.
   */

import { __exports as proptypes } from '../../../_virtual/proptypes.js';
export { __exports as default } from '../../../_virtual/proptypes.js';

Object.defineProperty(proptypes, "__esModule", { value: true });
proptypes.heightWidthRadiusDefaults = proptypes.heightWidthDefaults = proptypes.sizeMarginDefaults = proptypes.sizeDefaults = void 0;
/*
 * DefaultProps object for different loaders
 */
var commonValues = {
    loading: true,
    color: "#000000",
    css: "",
    speedMultiplier: 1
};
function sizeDefaults(sizeValue) {
    return Object.assign({}, commonValues, { size: sizeValue });
}
proptypes.sizeDefaults = sizeDefaults;
function sizeMarginDefaults(sizeValue) {
    return Object.assign({}, sizeDefaults(sizeValue), {
        margin: 2
    });
}
proptypes.sizeMarginDefaults = sizeMarginDefaults;
function heightWidthDefaults(height, width) {
    return Object.assign({}, commonValues, {
        height: height,
        width: width
    });
}
proptypes.heightWidthDefaults = heightWidthDefaults;
function heightWidthRadiusDefaults(height, width, radius) {
    if (radius === void 0) { radius = 2; }
    return Object.assign({}, heightWidthDefaults(height, width), {
        radius: radius,
        margin: 2
    });
}
proptypes.heightWidthRadiusDefaults = heightWidthRadiusDefaults;
