// @epignosis_llc/gnosis/dist/esm/src/components/Tabs/TabsNavItem.js

  /**
   * @license
   * author: epignosis front-end team
   * @epignosis_llc/gnosis.js v2.2.1
   * Released under the MIT license.
   */

import 'react';
import classNames from '../../../node_modules/classnames/index.js';
import { tabNavItem } from './styles.js';
import { createElement as jsx } from '../../../node_modules/@emotion/react/dist/emotion-react.browser.esm.js';

var TabsNavItem = function TabsNavItem(_ref) {
  var index = _ref.index,
      title = _ref.title,
      isActive = _ref.isActive,
      onSelectTab = _ref.onSelectTab;
  var HtmlClasses = classNames({
    "tab-link": true,
    selected: isActive
  });

  var onClick = function onClick(e) {
    e.preventDefault();
    onSelectTab(index);
  };

  return jsx("a", {
    css: function css(theme) {
      return tabNavItem(theme, {
        isActive: isActive
      });
    },
    href: "#tab-".concat(index),
    id: "tab-".concat(index),
    "aria-controls": "content-".concat(index),
    role: "tab",
    className: HtmlClasses,
    onClick: onClick
  }, title);
};

export { TabsNavItem as default };
