// @epignosis_llc/gnosis/dist/esm/src/components/Tooltip/Tooltip.js

  /**
   * @license
   * author: epignosis front-end team
   * @epignosis_llc/gnosis.js v2.2.1
   * Released under the MIT license.
   */

import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import 'react';
import Tippy from '../../../node_modules/@tippyjs/react/headless/dist/tippy-react-headless.esm.js';
import { tooltipContainer } from './styles.js';
import { createElement as jsx } from '../../../node_modules/@emotion/react/dist/emotion-react.browser.esm.js';

var _excluded = ["children", "content", "as", "placement"];

var Tooltip = function Tooltip(_ref) {
  var children = _ref.children,
      content = _ref.content,
      _ref$as = _ref.as,
      as = _ref$as === void 0 ? "div" : _ref$as,
      _ref$placement = _ref.placement,
      placement = _ref$placement === void 0 ? "top" : _ref$placement,
      rest = _objectWithoutProperties(_ref, _excluded);

  var Tag = as;
  return jsx(Tippy, _extends({
    interactive: true,
    placement: placement,
    render: function render(attrs) {
      return jsx("div", _extends({
        className: "tooltip",
        css: tooltipContainer
      }, attrs), content, jsx("div", {
        id: "arrow",
        "data-testid": "tooltip-arrow",
        "data-popper-arrow": true
      }));
    }
  }, rest), jsx(Tag, null, children));
};

export { Tooltip as default };
