// @epignosis_llc/gnosis/dist/esm/src/components/Heading/Heading.js

  /**
   * @license
   * author: epignosis front-end team
   * @epignosis_llc/gnosis.js v2.2.1
   * Released under the MIT license.
   */

import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import 'react';
import { headingContainer } from './styles.js';
import { createElement as jsx } from '../../../node_modules/@emotion/react/dist/emotion-react.browser.esm.js';

var _excluded = ["size", "as", "children"];
var SIZES = {
  h1: "3xl",
  h2: "2xl",
  h3: "xl",
  h4: "lg",
  h5: "md",
  h6: "sm"
};

var Heading = function Heading(_ref) {
  var size = _ref.size,
      _ref$as = _ref.as,
      as = _ref$as === void 0 ? "h2" : _ref$as,
      children = _ref.children,
      rest = _objectWithoutProperties(_ref, _excluded);

  var HeadingTag = as;
  var selectedSize = size || SIZES[as];
  return jsx(HeadingTag, _extends({
    css: function css(theme) {
      return headingContainer(theme, {
        size: selectedSize
      });
    }
  }, rest), children);
};

export { SIZES, Heading as default };
