// @epignosis_llc/gnosis/dist/esm/src/components/FormElements/RadioGroup/Radio.js

  /**
   * @license
   * author: epignosis front-end team
   * @epignosis_llc/gnosis.js v2.2.1
   * Released under the MIT license.
   */

import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../../../_virtual/_rollupPluginBabelHelpers.js';
import 'react';
import { radioButtonContainer } from './styles.js';
import { createElement as jsx } from '../../../../node_modules/@emotion/react/dist/emotion-react.browser.esm.js';

var _excluded = ["id", "label", "size", "inline", "containerAttrs"];

var Radio = function Radio(_ref) {
  var id = _ref.id,
      label = _ref.label,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? "md" : _ref$size,
      _ref$inline = _ref.inline,
      inline = _ref$inline === void 0 ? false : _ref$inline,
      containerAttrs = _ref.containerAttrs,
      rest = _objectWithoutProperties(_ref, _excluded);

  return jsx("div", _extends({
    css: function css(theme) {
      return radioButtonContainer(theme, {
        size: size,
        inline: inline
      });
    }
  }, containerAttrs), jsx("input", _extends({
    id: id,
    type: "radio"
  }, rest)), jsx("label", {
    htmlFor: id
  }, label));
};

export { Radio as default };
