// @epignosis_llc/gnosis/dist/esm/src/components/Pagination/Pagination.js

  /**
   * @license
   * author: epignosis front-end team
   * @epignosis_llc/gnosis.js v2.2.1
   * Released under the MIT license.
   */

import { extends as _extends } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import '../../../node_modules/core-js/modules/es.array.map.js';
import '../../../node_modules/core-js/modules/es.date.to-string.js';
import '../../../node_modules/core-js/modules/es.object.to-string.js';
import '../../../node_modules/core-js/modules/es.regexp.to-string.js';
import 'react';
import classNames from '../../../node_modules/classnames/index.js';
import Button from '../Button/Button.js';
import SvgArrowRight from '../../icons/arrow_right.svg.js';
import SvgArrowLeft from '../../icons/arrow_left.svg.js';
import { container } from './styles.js';
import { usePagination, ellipsis } from './usePagination.js';
import { createElement as jsx } from '../../../node_modules/@emotion/react/dist/emotion-react.browser.esm.js';

var Pagination = function Pagination(_ref) {
  var current = _ref.current,
      onChange = _ref.onChange,
      totalPages = _ref.totalPages,
      containerAttrs = _ref.containerAttrs;
  var paginationRange = usePagination(current, totalPages);

  var classNamesContainer = function classNamesContainer(pageNumber) {
    return classNames({
      isActive: pageNumber === current
    });
  };

  return jsx("div", _extends({
    css: container
  }, containerAttrs), jsx(Button, {
    className: "previous-page-btn",
    "data-testid": "previous-page-btn",
    name: "Previous page",
    onClick: function onClick() {
      return onChange(current - 1);
    },
    color: "secondary",
    noGutters: true,
    disabled: current === 1
  }, jsx(SvgArrowLeft, {
    height: 22
  })), jsx("div", {
    className: "pagination-options"
  }, paginationRange.map(function (pageNumber) {
    if (pageNumber === ellipsis) {
      return ellipsis;
    }

    return jsx(Button, {
      type: "button",
      key: pageNumber,
      id: pageNumber.toString(),
      "data-testid": "pagination-page",
      onClick: function onClick() {
        return onChange(pageNumber);
      },
      variant: "ghost",
      noGutters: true,
      className: classNamesContainer(pageNumber)
    }, pageNumber);
  })), jsx(Button, {
    className: "next-page-btn",
    "data-testid": "next-page-btn",
    name: "Next page",
    onClick: function onClick() {
      return onChange(current + 1);
    },
    color: "secondary",
    noGutters: true,
    disabled: current === totalPages
  }, jsx(SvgArrowRight, {
    height: 22
  })));
};

export { Pagination as default };
