// @epignosis_llc/gnosis/dist/esm/src/components/MediaPlayer/AudioPlayer/AudioPlayer.js

  /**
   * @license
   * author: epignosis front-end team
   * @epignosis_llc/gnosis.js v2.2.1
   * Released under the MIT license.
   */

import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../../../_virtual/_rollupPluginBabelHelpers.js';
import 'react';
import ReactPlayer from '../../../../node_modules/react-player/lazy/index.js';
import { createElement as jsx } from '../../../../node_modules/@emotion/react/dist/emotion-react.browser.esm.js';

var _excluded = ["src", "controls", "containerAttrs"];

var AudioPlayer = function AudioPlayer(_ref) {
  var src = _ref.src,
      _ref$controls = _ref.controls,
      controls = _ref$controls === void 0 ? true : _ref$controls,
      containerAttrs = _ref.containerAttrs,
      rest = _objectWithoutProperties(_ref, _excluded);

  return jsx("div", containerAttrs, jsx(ReactPlayer, _extends({
    url: src,
    controls: controls,
    className: "react-player",
    height: "3.5rem",
    width: "100%"
  }, rest)));
};

export { AudioPlayer as default };
