// @epignosis_llc/gnosis/dist/esm/src/components/Card/Drawer.js

  /**
   * @license
   * author: epignosis front-end team
   * @epignosis_llc/gnosis.js v2.2.1
   * Released under the MIT license.
   */

import { objectSpread2 as _objectSpread2 } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import 'react';
import '../../../node_modules/@umijs/use-request/es/index.js';
import '../../../node_modules/lodash.isequal/index.js';
import '../../../node_modules/resize-observer-polyfill/dist/ResizeObserver.es.js';
import useClickAway from '../../../node_modules/@umijs/hooks/es/useClickAway/index.js';
import '../../../node_modules/screenfull/dist/screenfull.js';
import '../../../node_modules/intersection-observer/intersection-observer.js';
import '../../../node_modules/@umijs/hooks/es/useTextSelection/index.js';
import SvgClose from '../../icons/close.svg.js';
import { drawerContainer } from './styles.js';
import { createElement as jsx } from '../../../node_modules/@emotion/react/dist/emotion-react.browser.esm.js';
import { AnimatePresence } from '../../../node_modules/framer-motion/dist/es/components/AnimatePresence/index.mjs.js';
import { m } from '../../../node_modules/framer-motion/dist/es/render/dom/motion-minimal.mjs.js';

var transition = {
  duration: 0.25,
  ease: "easeInOut"
};
var drawerContainerVariants = {
  open: {
    width: "100%",
    transition: _objectSpread2(_objectSpread2({}, transition), {}, {
      when: "beforeChildren"
    }),
    transitionEnd: {
      overflowY: "auto"
    }
  },
  closed: {
    width: "0%",
    overflowY: "hidden",
    transition: _objectSpread2(_objectSpread2({}, transition), {}, {
      when: "afterChildren"
    })
  }
};
var drawerContentVariants = {
  open: {
    opacity: 1
  },
  closed: {
    opacity: 0
  }
};

var Drawer = function Drawer(_ref) {
  var isOpen = _ref.isOpen,
      onClose = _ref.onClose,
      title = _ref.title,
      footer = _ref.footer,
      children = _ref.children;
  var containerRef = useClickAway(function () {
    if (isOpen) {
      onClose();
    }
  });
  return jsx(AnimatePresence, null, isOpen && jsx(m.section, {
    css: drawerContainer,
    ref: containerRef,
    initial: "closed",
    animate: isOpen && "open",
    exit: "closed",
    variants: drawerContainerVariants
  }, jsx(m.div, {
    className: "content-wrapper",
    variants: drawerContentVariants
  }, jsx("div", null, jsx("header", null, jsx("div", null, title), jsx("button", {
    onClick: onClose
  }, jsx(SvgClose, {
    height: 16
  }))), jsx("p", null, children)), footer && jsx("footer", null, footer))));
};

export { Drawer as default };
