// @epignosis_llc/gnosis/dist/esm/src/theme/utils/breakpoints.js

  /**
   * @license
   * author: epignosis front-end team
   * @epignosis_llc/gnosis.js v2.2.1
   * Released under the MIT license.
   */

import { objectSpread2 as _objectSpread2, defineProperty as _defineProperty } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import '../../../node_modules/core-js/modules/es.array.reduce.js';
import '../../../node_modules/core-js/modules/es.object.to-string.js';
import '../../../node_modules/core-js/modules/es.object.keys.js';
import '../../../node_modules/@umijs/use-request/es/index.js';
import 'react';
import '../../../node_modules/lodash.isequal/index.js';
import '../../../node_modules/resize-observer-polyfill/dist/ResizeObserver.es.js';
import { configResponsive } from '../../../node_modules/@umijs/hooks/es/useResponsive/index.js';
import '../../../node_modules/screenfull/dist/screenfull.js';
import '../../../node_modules/intersection-observer/intersection-observer.js';
import '../../../node_modules/@umijs/hooks/es/useTextSelection/index.js';

var breakpoints = {
  xs: 320,
  sm: 576,
  md: 768,
  lg: 992,
  xl: 1200,
  xxl: 1600
};
var mq = Object.keys(breakpoints).reduce(function (acc, breakpointName) {
  return _objectSpread2(_objectSpread2({}, acc), {}, _defineProperty({}, breakpointName, "@media screen and (min-width: ".concat(breakpoints[breakpointName], "px)")));
}, {}); // Configures useResponsive hook

configResponsive(breakpoints);

export { breakpoints, mq };
