// @epignosis_llc/gnosis/dist/esm/src/components/Badge/Badge.js

  /**
   * @license
   * author: epignosis front-end team
   * @epignosis_llc/gnosis.js v2.2.1
   * Released under the MIT license.
   */

import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import 'react';
import classNames from '../../../node_modules/classnames/index.js';
import { container } from './styles.js';
import { createElement as jsx } from '../../../node_modules/@emotion/react/dist/emotion-react.browser.esm.js';

var _excluded = ["size", "offset", "children", "badgeContent", "containerAttrs"];

var Badge = function Badge(_ref) {
  var _ref$size = _ref.size,
      size = _ref$size === void 0 ? "md" : _ref$size,
      _ref$offset = _ref.offset,
      offset = _ref$offset === void 0 ? {
    top: "-8px",
    right: "-15px"
  } : _ref$offset,
      children = _ref.children,
      badgeContent = _ref.badgeContent,
      containerAttrs = _ref.containerAttrs,
      rest = _objectWithoutProperties(_ref, _excluded);

  var containerClassNames = classNames({
    "has-content": badgeContent,
    "big-content": badgeContent && badgeContent.length >= 3
  });
  return jsx("div", _extends({
    css: function css(theme) {
      return container(theme, {
        size: size,
        offset: offset
      });
    },
    className: containerClassNames
  }, containerAttrs), jsx("span", _extends({
    className: "content-container"
  }, rest), badgeContent), children);
};

export { Badge as default };
