// @epignosis_llc/gnosis/dist/esm/src/theme/default/colors.js

  /**
   * @license
   * author: epignosis front-end team
   * @epignosis_llc/gnosis.js v2.2.1
   * Released under the MIT license.
   */

import '../../../node_modules/core-js/modules/es.parse-int.js';
import '../../../node_modules/core-js/modules/es.array.slice.js';
import '../../../node_modules/core-js/modules/es.array.concat.js';
import color from '../../../node_modules/color/index.js';

var THEME_COLOURS = {
  blue: "#1B68B3",
  turquoise: "#2CD1F8",
  orange: "#FF9C28",
  red: "#F20000",
  green: "#41BD54",
  gray: "#6E6E6E",
  grayBlue: "#9EA5A9",
  black: "#232323",
  white: "#FFFFFF"
};
var palletGenerator = function palletGenerator(base) {
  return {
    base: base,
    lightest: color(base).lighten(0.7).hex(),
    lighter: color(base).lighten(0.5).hex(),
    light: color(base).lighten(0.2).hex(),
    dark: color(base).darken(0.2).hex(),
    darker: color(base).darken(0.4).hex(),
    darkest: color(base).darken(0.6).hex()
  };
};
var colors = {
  blue: palletGenerator(THEME_COLOURS.blue),
  turquoise: palletGenerator(THEME_COLOURS.turquoise),
  orange: palletGenerator(THEME_COLOURS.orange),
  red: palletGenerator(THEME_COLOURS.red),
  green: palletGenerator(THEME_COLOURS.green),
  gray: palletGenerator(THEME_COLOURS.gray),
  grayBlue: palletGenerator(THEME_COLOURS.grayBlue),
  black: THEME_COLOURS.black,
  white: THEME_COLOURS.white
};
var hexToRGBA = function hexToRGBA(hex, alpha) {
  var red = parseInt(hex.slice(1, 3), 16);
  var green = parseInt(hex.slice(3, 5), 16);
  var blue = parseInt(hex.slice(5, 7), 16);
  return "rgba(".concat(red, ", ").concat(green, ", ").concat(blue, ", ").concat(alpha, ")");
};

export { THEME_COLOURS, colors, hexToRGBA, palletGenerator };
