// @epignosis_llc/gnosis/dist/esm/src/theme/default/defaultTheme.js

  /**
   * @license
   * author: epignosis front-end team
   * @epignosis_llc/gnosis.js v2.2.1
   * Released under the MIT license.
   */

import { generateTypeScaleSizes, DEFAULT_TYPESCALE_CONFIG } from '../utils/typography.js';
import { colors } from './colors.js';
import alert from './config/alert.js';
import badge from './config/badge.js';
import breadcrumb from './config/breadcrumb.js';
import button from './config/button.js';
import card from './config/card.js';
import chip from './config/chip.js';
import drawer from './config/drawer.js';
import formElements from './config/formElements.js';
import loader from './config/loader.js';
import modal from './config/modal.js';
import progressBar from './config/progressBar.js';
import result from './config/result.js';
import sidebar from './config/sidebar.js';
import tabs from './config/tabs.js';
import tag from './config/tag.js';
import tooltip from './config/tooltip.js';
import pagination from './config/pagination.js';

var typeScaleSizes = generateTypeScaleSizes(DEFAULT_TYPESCALE_CONFIG);
var defaultTheme = {
  body: {
    color: "#232323",
    background: colors.white,
    fontSize: "100%",
    lineHeight: 1.5715,
    fontFamily: "\"Mulish\", Arial, sans-serif"
  },
  link: {
    color: colors.blue.base,
    hoverColor: colors.blue.lighter
  },
  typeScaleSizes: typeScaleSizes,
  alert: alert,
  badge: badge,
  breadcrumb: breadcrumb,
  button: button,
  card: card,
  chip: chip,
  drawer: drawer,
  formElements: formElements,
  loader: loader,
  modal: modal,
  progressBar: progressBar,
  result: result,
  sidebar: sidebar,
  tabs: tabs,
  tag: tag,
  tooltip: tooltip,
  pagination: pagination
};

export { defaultTheme as default };
